### ISSUE TITLE
Find a better example for good output than `git push`

### ISSUE THREAD

**Issue by aanand** [cli-guidelines/cli-guidelines#43]
`git push` says a bit too much. What's a better example of a command that's explicit without being too wordy?

I think it should be something that communicates several pieces of information (not just a trivial one-liner).

**Comment by fwilhe2** [comment]
I think the challenge here is that developers often consider things important that users don't want or need to care about. But also what users really want or need to know might be different depending on what they are doing.

I would suggest `docker pull` as a good example. Consider this "screenshot":

```
$ docker pull wordpress
Using default tag: latest
latest: Pulling from library/wordpress
852e50cd189d: Already exists 
0266fc315b01: Pull complete 
4c8a5fa787a1: Downloading [==============================>                    ]   47.3MB/76.65MB
46fc127c1884: Download complete 
f768b7fadf16: Download complete 
345b578c1a78: Download complete 
90aafe41e78d: Download complete 
0b5d7691301b: Download complete 
e06cc2d9e2f8: Download complete 
6e9815fb33e7: Download complete 
9c620d71744c: Download complete 
66ef24dcaca4: Download complete 
83f52dab9487: Download complete 
83626a606b22: Download complete 
81fb279f13d3: Download complete 
1417d770ba43: Download complete 
3831bd0339c0: Download complete 
bf496d74576e: Download complete 
cc074193f10d: Downloading [==============================================>    ]   11.9MB/12.85MB
21692721b48c: Download complete 
```

What makes this good imo is that it provides me a sense of progress for something that might take a second, or very long depending on how fast the download is and how big the image is. Sure, you might not need to know all the layers, but I still like how Docker makes me feel in control of "is this going to take a minute or half an hour?".

**Comment by aanand** [comment]
I agree that `docker pull` is a good example of a command that lets me know what it's doing. However (and I should have provided this context in the first place, sorry), this is the advice point where we use `git push`:

> **If you change state, tell the user.**
When a command changes the state of a system, it’s especially valuable to explain what has just happened, so the user can model the state of the system in their head—particularly if the result doesn’t directly map to what the user requested.
>
> For example, `git push` tells you exactly what it is doing, and what the new state of the remote branch is:
>
> ```
> $ git push
> Enumerating objects: 18, done.
> Counting objects: 100% (18/18), done.
> Delta compression using up to 8 threads
> Compressing objects: 100% (10/10), done.
> Writing objects: 100% (10/10), 2.09 KiB | 2.09 MiB/s, done.
> Total 10 (delta 8), reused 0 (delta 0), pack-reused 0
> remote: Resolving deltas: 100% (8/8), completed with 8 local objects.
> To github.com:replicate/replicate.git
> + 6c22c90...a2a5217 bfirsh/fix-delete -> bfirsh/fix-delete
> ```

I think that there are two issues with `git push` as an example here. Firstly, as we know, it's too verbose - all that stuff about `Delta compression` and `pack-reused` is completely irrelevant to 99% of users.

Secondly, though, it's a bit simplistic? The point of `git push` is to update the state of a branch on a remote. The useful bit of `git push`'s output is where it tells you it did that. Not much of a bar to clear.

I think a more interesting example would involve a command that perhaps changes _several pieces_ of system state. Or one that might change some state as a possibly-unexpected side effect.

**Comment by strokirk** [comment]
The various [`kubectl`](https://kubernetes.io/docs/reference/kubectl/cheatsheet/) commands tend to update state in various ways, and usually have fairly simple output.
